src/java/htsjdk/samtools/cram/build/Sam2CramRecordFactory.java

```diff
@@ -96,7 +99,17 @@ public Sam2CramRecordFactory(final byte[] refBases, final SAMFileHeader samFileH
         }
     }
 
+    /**
+     * Create a CramCompressionRecord.
+     *
+     * @param record If the input record does not have an associated SAMFileHeader, it will be updated
+     *               with the header used for the writer in order to allow reference indices to be resolved.
```

Comment:
Technically, no, though it only gets used to create records for CRAMFileWriter. Still, its confusing to use the word "writer" here so I'll change it to "factory."